txscript/opcode.go

```diff
@@ -2051,7 +2061,18 @@ func opcodeCheckMultiSig(op *parsedOpcode, vm *Engine) error {
 		// Generate the signature hash based on the signature hash type.
 		hash := calcSignatureHash(script, hashType, &vm.tx, vm.txIdx)
 
-		if parsedSig.Verify(hash, parsedPubKey) {
+		var valid bool
+		if vm.sigCache != nil {
+			valid = vm.sigCache.Exists(hash, parsedSig, parsedPubKey)
```

Comment:
The same is true here as above regarding existence signalling the signature is valid.